packages/assistify-dbs-translation/server/dbsTranslate.js

```diff
@@ -0,0 +1,218 @@
+	_sendRequestTranslateMessage(message, targetLanguages) {
+		const translations = {};
+		const msgs = message.msg.split('\n');
+		const query = msgs.join();
+		let sourceLanguage;
+		/**
+		 * Service provider do not handle the text language detection automatically rather it requires the source language to be specified
+		 * explicitly. To automate this language detection process we used the cld language detector.
+		 * When the language detector fails, it falls back into the user language.
+		 */
+		cld.detect(query, (err, result) => {
+			if (result) {
+				result.languages.map((language) => {
+					sourceLanguage = language.code;
+				});
+				const supportedLanguages = this._getSupportedLanguages('en');
+				targetLanguages.forEach(language => {
+					if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+						language = language.substr(0, 2);
+					}
+					const result = Meteor.call('assistify.translate', 'POST', `${ this.apiEndPointUrl }/translate`, {
+						params: {
+							key: this.apiKey
+						}, data: {
+							text: query,
+							to: language,
+							from: sourceLanguage
+						}
+					});
+					translations[language] = this.deTokenize(Object.assign({}, message, {msg: result}));
+				});
+			} else {
+				SystemLogger.log('Text language could not be determined', err.message);
+			}
+		});
+		return translations;
+	}
+
+	/**
+	 * Returns translated message attachment description in target languages.
+	 * @private
+	 * @param {object} attachment
+	 * @param {object} targetLanguages
+	 * @returns {object} translated messages for each target language
+	 */
+	_sendRequestTranslateMessageAttachments(attachment, targetLanguages) {
+		const translations = {};
+		const query = attachment.description || attachment.text;
+		let sourceLanguage;
+		/**
+		 * Service provider do not handle the text language detection automatically rather it requires the source language to be specified
+		 * explicitly. To automate this language detection process we used the cld language detector.
+		 * When the language detector fails, it falls back into the user language.
+		 */
+		cld.detect(query, (err, result) => {
+			if (result) {
+				result.languages.map((language) => {
+					sourceLanguage = language.code || RocketChat.settings.get('Language'); // fallback to user language.
+				});
+				const supportedLanguages = this._getSupportedLanguages('en');
+				targetLanguages.forEach(language => {
+					if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+						language = language.substr(0, 2);
+					}
+					translations[language] = Meteor.call('assistify.translate', 'POST', `${ this.apiEndPointUrl }/translate`, {
+						params: {
+							key: this.apiKey
+						}, data: {
+							text: query,
+							to: language,
+							from: sourceLanguage
+						}
+					});
+				});
+			} else {
+				SystemLogger.log('Text language could not be determined', err.message);
+			}
+		});
+		return translations;
```

Comment:
this return is outside the callback => should this be `await`ed (return a Promise)?